fastq_filtrator.py

```diff
@@ -0,0 +1,199 @@
+def fastq_filter(gc_bounds, length_bounds, quality_threshold,
+                 save_filtered, input_fastq, output_file_prefix):
+    sequence_dictionary, sequence_dictionary_original = open_fastq_file(input_fastq)
+    sequence_dictionary = quality_control(sequence_dictionary)
+    parameters = input('Do you want to change any parameters? Please type Y/N.\n By default:\n\n'
+                       '* gc_bounds = (0,100)\n'
+                       '* length_bounds = (0,2**32)\n'
+                       '* quality_threshold = 0\n'
+                       '* save_filtered = False\n\n')
+    if parameters == 'Y':
+        gc_bounds, length_bounds, quality_threshold, save_filtered = change_of_parameters(gc_bounds,
+                                                                                          length_bounds,
+                                                                                          quality_threshold,
+                                                                                          save_filtered)
+    else:
+        pass
+    sequence_dictionary, list_to_be_removed = gc_content_filter(sequence_dictionary, gc_bounds)
+    sequence_dictionary, list_to_be_removed = length_filter(sequence_dictionary, length_bounds, list_to_be_removed)
+    sequence_dictionary, list_to_be_removed = quality_filter(sequence_dictionary, quality_threshold, list_to_be_removed)
+    saving(sequence_dictionary, sequence_dictionary_original, output_file_prefix, save_filtered, list_to_be_removed)
+
+
+def open_fastq_file(input_fastq):
+    with open(input_fastq, 'r') as file:
+        fastq_file = file.readlines()
+    sequence_dictionary = {}
+    sequence_dictionary_original = {}
+    for i in range(0, len(fastq_file)-1, 4):
+        id_code = fastq_file[i].replace("\n", "")
+        sequence = fastq_file[i+1].replace("\n", "")
+        third_line = fastq_file[i+2].replace("\n", "")
+        quality_score = fastq_file[i+3].replace("\n", "")
+        sequence_dictionary[i//4] = [id_code, sequence, third_line, quality_score]
+    sequence_dictionary_original = sequence_dictionary
+    return sequence_dictionary, sequence_dictionary_original
+
+
+def quality_control(sequence_dictionary):
+    length_of_reads = []
+    gc_values = []
+    quality_values = []
+    for key in sequence_dictionary:
+        sequence = sequence_dictionary[key][1]
+        gc_composition = ((sequence.count('G')+sequence.count('C'))/len(sequence))*100
+        gc_values.append(round(gc_composition, 1))
+        sequence_dictionary[key].append(round(gc_composition, 1))
+        length_of_reads.append(len(sequence))
+        sequence_dictionary[key].append(len(sequence))
+        quality = []
+        for i in sequence_dictionary[key][3]:
+            quality.append(ord(i)-33)
+        sequence_dictionary[key].append(numpy.mean(quality))
```

Comment:
Обычно делают `import numpy as np`
```suggestion
        sequence_dictionary[key].append(np.mean(quality))
```